lms/djangoapps/philu_api/views.py

```diff
@@ -162,6 +163,50 @@ def post(self, request):
         return JsonResponse({'success': False}, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
 
 
+class GetRemainingBadges(APIView):
+    def get(self, request):
+        user_id = request.user.id
+        community_id = request.GET.get('community_id')
+
+        try:
+            remaining_badges = Badge.get_remaining_badges(user_id=user_id,
+                                                          community_id=community_id)
+            if remaining_badges:
+                remaining_badges_json = {}
+                for badge in remaining_badges:
+                    remaining_badges_json[badge.id] = {'name': badge.name,
+                                                       'description': badge.description,
+                                                       'threshold': badge.threshold,
+                                                       'type': badge.type,
+                                                       'image': badge.image}
+
+                return JsonResponse({'success': True, 'badges': remaining_badges_json})
+            else:
+                return JsonResponse({'success': True, 'badges': {}})
+        except Exception as e:
+            return JsonResponse({'success': False, 'message': str(e)})
+
+
+class AssignUserBadge(APIView):
+    def post(self, request):
+        user_id = request.data.get('user_id')
+        badge_id = request.data.get('badge_id')
+        community_id = request.data.get('community_id')
+        master_token = request.GET.get('token')
```

Comment:
will you get token from GET in post call?

Reply:
NodeBB is sending token in query params. According to @arbisofttaha that's how every other request is doing it